sdk/tests/deqp/modules/shared/glsDrawTests.js

```diff
@@ -1524,228 +1475,56 @@ goog.scope(function() {
-
-                        vertex2 = generateVertex(vertex1, vertexParameters);
-                        vertex2[parallelaxis] = vertex1[parallelaxis];
-
-                        vertex3 = generateVertex(vertex2, vertexParameters);
-                        vertex3[oppositeaxis] = vertex1[oppositeaxis];
-
-                        direction = packed ?
-                            (vertex3[parallelaxis] > vertex1[parallelaxis] ? 1 : -1) :
-                            (vertex3[parallelaxis].greaterThan(vertex1[parallelaxis]) ? 1 : -1);
-
-                        vertex4 = generateVertex(vertex3, vertexParameters);
-                        vertex4[oppositeaxis] = vertex2[oppositeaxis];
-                        vertex4[parallelaxis] = vertex3[parallelaxis];
-
-                        quadVertices = [vertex1, vertex2, vertex3, vertex4];
+            var components = [];
+
+            for (var componentNdx = 0; componentNdx < componentCount; componentNdx++) {
+                var getRandomComponent = function() {
+                    // For packed formats we can't use GLValue
+                    if (packed) {
+                        if (componentNdx == 3) {
+                            return rnd.getInt() % limit2;
+                        } else {
+                            return rnd.getInt() % limit10;
+                        }
                     } else {
-                        vertex5 = generateVertex(components, vertexParameters);
-                        vertex5[oppositeaxis] = quadVertices[quadVertices.length - 2][oppositeaxis];
-                        if (!packed)
-                            vertex5[parallelaxis] = glsDrawTests.GLValue.getRandom(
-                                vertexParameters.rnd,
-                                direction > 0 ? components[parallelaxis] : vertexParameters.min,
-                                direction > 0 ? vertexParameters.max : components[parallelaxis]
-                            );
-
-                        vertex6 = generateVertex(vertex5, vertexParameters);
-                        vertex6[oppositeaxis] = quadVertices[quadVertices.length - 1][oppositeaxis];
-                        vertex6[parallelaxis] = vertex5[parallelaxis];
-
-                        quadVertices = [vertex5, vertex6];
+                        return glsDrawTests.GLValue.getRandom(rnd, min, max);
                     }
-                    break;
-                }
-                case glsDrawTests.DrawTestSpec.Primitive.TRIANGLE_FAN: {
-                    if (vertexNdx < first + 4) {
-                        vertex1 = generateVertex(components, vertexParameters);
+                };
 
-                        parallelaxis = rnd.getBool() ? 0 : 1;
-                        oppositeaxis = parallelaxis ? 0 : 1;
+                var component = getRandomComponent();
+                var componentFloat = (component instanceof glsDrawTests.GLValue) ? component.toFloat() : component;
```

Comment:
The dEQP code doesn't use componentFloat and uses multiple loops on componentNdx. The logic here is he same but expressed slightly differently.